internal/install/segment/client.go

```diff
@@ -0,0 +1,95 @@
+package segment
+
+import (
+	"encoding/json"
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/segmentio/analytics-go.v3"
+)
+
+type EventType string
+
+var EventTypes = struct {
+	InstallStarted          EventType
+	AccountIDMissing        EventType
+	APIKeyMissing           EventType
+	RegionMissing           EventType
+	UnableToConnect         EventType
+	UnableToFetchLicenseKey EventType
+	LicenseKeyFetchedOk     EventType
+	UnableToOverrideClient  EventType
+}{
+	InstallStarted:          "InstallStarted",
+	AccountIDMissing:        "AccountIDMissing",
+	APIKeyMissing:           "APIKeyMissing",
+	RegionMissing:           "RegionMissing",
+	UnableToConnect:         "UnableToConnect",
+	UnableToFetchLicenseKey: "UnableToFetchLicenseKey",
+	LicenseKeyFetchedOk:     "LicenseKeyFetchedOk",
+	UnableToOverrideClient:  "UnableToOverrideClient",
+}
+
+type Segment struct {
+	analytics.Client
+}
+
+func New(writeKey string) *Segment {
```

Comment:
@NRhzhao what if we're injecting the additional fields we always want to track event during init, for example accountID and region? Could we maybe get them from os.GetEnv( ). Note for region, we should also handle the case when empty means US.
That will make analyzing those events quite easier.
This will also lighten up the contract of the Track( ) method.